addons/udes_stock/models/abstract_stock_model.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+from odoo.osv import expression
+
+
+class AbstractStockModel(models.AbstractModel):
+    _name = 'abstract.stock.model'
+
+    # Whether the model can create or not
+    CREATE = False
+    # The default domain to search with a str
+    STR_DOMAIN = ['name']
+
+    def get_or_create(self, identifier, create=False):
+        """ Get or create in model, for create also need CREATE to be True
+        :args:
+            - identifier: str or int
+                The identifier to search with
+        :kwargs:
+            - create: Boolean
+                If true, and CREATE is true, then a new object is created
+        :returns:
+            The model queried
+        """
+        model = self._name
+        model_type = model.split('.')[-1]
+        # Get the domain
+        if isinstance(identifier, str):
+            domain = expression.OR([[(field, '=', identifier)] for field in self.STR_DOMAIN])
+        elif isinstance(identifier, int):
+            domain = [('id', '=', identifier)]
+        else:
+            raise ValidationError(
+                _(f'Identifier must be either int or str, not {type(identifier)}')
+            )
+        results = self.search(domain)
+
+        if not results:
+            if self.CREATE and create and isinstance(identifier, str):
+                results = self.create({'name': identifier})
+            elif self.CREATE and create:
+                raise ValidationError(
+                    _(
+                        f'Cannot create a new {model_type} for {model} with identifier of type {type(identifier)}'
+                    )
+                )
+            elif create:
+                raise ValidationError(_(f'Cannot create a new {model_type} for {model}'))
+            else:
+                raise ValidationError(_(f'{model_type} not found for identifier {identifier}'))
+        elif len(results) > 1:
+            raise ValidationError(
+                _(f'Too many {model_type}s found for identifier {identifier} in {model}')
+            )
+        elif create:
```

Comment:
This case is not needed, the intention of the function is to get the object and if it is not found create it.